component/azstorage/block_blob.go

```diff
@@ -241,18 +241,22 @@ func (bb *BlockBlob) SetPrefixPath(path string) error {
 func (bb *BlockBlob) CreateFile(name string, mode os.FileMode) error {
 	log.Trace("BlockBlob::CreateFile : name %s", name)
 	var data []byte
-	return bb.WriteFromBuffer(name, nil, data)
+	return bb.WriteFromBuffer(internal.WriteFromBufferOptions{Name: name,
+		Data: data})
 }
 
 // CreateDirectory : Create a new directory in the container/virtual directory
-func (bb *BlockBlob) CreateDirectory(name string) error {
+func (bb *BlockBlob) CreateDirectory(name string, etag bool) error {
 	log.Trace("BlockBlob::CreateDirectory : name %s", name)
 
 	var data []byte
 	metadata := make(map[string]*string)
 	metadata[folderKey] = to.Ptr("true")
 
-	return bb.WriteFromBuffer(name, metadata, data)
+	return bb.WriteFromBuffer(internal.WriteFromBufferOptions{Name: name,
```

Comment:
If ETAG is nil, where user does not want to compare then shall we even send the etag in REST call (we were not sending it earlier)